sql/conn_handler/socket_connection.cc

```diff
@@ -1471,7 +1471,14 @@ void Mysqld_socket_listener::close_listener() {
     being listened is not included in the m_socket_map. Instead, this socket
     referenced by the data member m_admin_interface_listen_socket.
   */
-  if (m_use_separate_thread_for_admin) {
+
+  /*
+    See the logic in setup_listener()
+    Thread was spawned only if both --admin-address is specified
+    and --create-admin-listener-thread=ON
+   */
+  if (!m_admin_bind_address.address.empty() &&
+      m_use_separate_thread_for_admin) {
```

Comment:
Have you verified that commands executed on the separate admin thread are replicated correctly?

Reply:
Yes, it works fine. Admin connection differs from the normal user connection only in the process of establishing connection.
1. We have dedicated connection accepting thread
2. There is no limitation for number of connections
So once the connection is established, it is processed by exactly the same mechanism as normal user connection.